sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -267,6 +267,11 @@ func resourceSumologicMonitorsLibraryMonitor() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{"Logs", "Metrics"}, false),
 			},
 
+			"evaluation_delay": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
You might want to add a `ValidateFunc` here for input validation. See `time_range` for an example.

Reply:
I realize you might not be able to just use `time_range`'s validation function here as you probably want to admit a broader set of values. 

If that's the case, see if you could express it as a regex. You could then hand-code a validation function as done [here](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#validatefunc).

Reply:
There is lot of validation code (Monitor manager validators) for the delay values permitted. When we try to create a monitor using terraform, won't that validation code return error messages to user? cc @rsanbhad

Reply:
or maybe no, because those are REST API messages?

Reply:
those error messages will happen on `terraform apply`, but I think Avijit is asking if we can do similar validations here - if you validate in the terraform provider, then those validations run on `terraform plan` which is a better user experience (it was raised in the past that monitors terraform could do more validation locally rather than relying on server validation errors, as those make it more difficult for users to know if their terraform configuration is correct when writing it)

Reply:
i have added ValidateFunc

Reply:
nice!